### PR TITLE
Add exclusion list, multiple load modes, update UI copy

### PR DESCRIPTION
**Pull request by aidvu**
- Add the ability to choose JS execution mode for concatenated scripts: async or defer.
- Add an option with a comma separated list for exclusion of scripts by searching for the substring(s) given.

### CODE REVIEW

File: concat-js.php [hunk 1]
```diff
@@ -52,11 +52,9 @@ function do_items( $handles = false, $group = false ) {
 		$handles = false === $handles ? $this->queue : (array) $handles;
 		$javascripts = array();
 		$siteurl = apply_filters( 'page_optimize_site_url', $this->base_url );
-
 		$this->all_deps( $handles );
 		$level = 0;
 
-		$total_count = count( $this->to_do );
```
Comment (aidvu) [inline concat-js.php:59]: unused

File: concat-js.php [hunk 2]
```diff
@@ -84,7 +82,6 @@ function do_items( $handles = false, $group = false ) {
 			$obj = $this->registered[ $handle ];
 			$js_url = $obj->src;
 			$js_url_parsed = parse_url( $js_url );
-			$extra = $obj->extra;
```
Comment (aidvu) [inline concat-js.php:87]: unused

File: concat-js.php [hunk 3]
```diff
@@ -240,26 +240,12 @@ function do_items( $handles = false, $group = false ) {
 				if ( isset( $href ) ) {
 					$handles = implode( ',', $js_array['handles'] );
 
-					$load_mode = '';
-					if ( page_optimize_should_defer_noncritcal_js() ) {
-						$load_mode = "defer";
-
-						// Stuff to skip loading JS async/defer
+					$load_mode = page_optimize_load_mode_js();
```
Comment (brandonpayton) [inline concat-js.php:243]: I like this change. More flexible.

File: settings.php [hunk 4]
```diff
@@ -18,39 +18,64 @@ function page_optimize_settings_page() {
 }
 
 function page_optimize_settings_section() {
-	// TODO: Add description
+	_e( 'Concat JavaScript and CSS for faster site loading and lower number of requests. Scripts are grouped by the original placement.' );
```
Comment (brandonpayton) [inline settings.php:21]: I am still getting familiar with WordPress. Why use `_e()` instead of `__()`?

Comment (aidvu) [reply settings.php:21]: `_e()` == `echo __()`

Comment (brandonpayton) [reply settings.php:21]: don't know how I missed that. thank you

**Pull request merged**